test/spin-up-test-environment.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash -e
+
+PUB_KEY=$(cat ~/.rsa_key.pub)
+gateway=$(ip r | grep default | awk '{print $3}')
+
+function create_vm() {
+    local name=$1
+    local port=$2
+    local ip=$3
+    local additional_forwarding=$4
+
+    echo "#cloud-config" > .temp/user-data
+    echo "" >> .temp/user-data
+    echo "#cloud-config" > .temp/network
+    echo "" >> .temp/network
+
+    cat cloud-init/user-data | yq ".hostname = \"${name}\" | .fqdn = \"${name}.k8s.local\" | .users[0].ssh_authorized_keys = [\"${PUB_KEY}\"]" --yaml-output >> .temp/user-data
+    cat cloud-init/network | yq --yaml-output ".network.ethernets.enp0s3.addresses += [\"10.255.254.${ip}/24\"]" >> .temp/network
+
+    cloud-localds .temp/cloud-init-${name}.iso .temp/user-data -N .temp/network > /dev/null
+
+    echo "Creating VM $name on port $port..."
+    [ -f .temp/${name}.img ] && rm .temp/${name}.img
+    qemu-img create -b ubuntu.img -F qcow2 -f qcow2 .temp/${name}.img 20G
+
+    sudo -b qemu-system-x86_64 \
+        -enable-kvm \
+        -cpu host \
+        -boot menu=off \
+        -drive if=pflash,format=raw,readonly=on,file=.temp/OVMF_CODE_4M.fd \
+        -drive if=pflash,format=raw,file=.temp/OVMF_VARS_4M.fd \
+        -drive file=.temp/${name}.img \
+        -cdrom .temp/cloud-init-${name}.iso \
+        -device virtio-net-pci,netdev=net0,mac=52:54:00:00:00:${ip} \
+        -device virtio-net-pci,netdev=net1,mac=52:54:00:00:01:${ip} \
+        -netdev user,id=net0,hostfwd=tcp::${port}-:22${additional_forwarding} \
+        -netdev socket,id=net1,mcast=230.0.0.1:1234 \
+        -m 4G \
+        -machine accel=kvm,type=q35 \
+        -smp 2 \
+        -nographic \
+        -smbios type=0,uefi=on 1>.temp/${name}.stdout.log 2>.temp/${name}.stderr.log
+
+  sleep 3
+}
+
+function wait_for_ssh() {
+  local host=$1
+  local port=$2
+  echo "Waiting for SSH on $host:$port..."
```

Comment:
To be consistent with the rest of the code there should be curley braces.

```suggestion
  echo "Waiting for SSH on ${host}:${port}..."
```